Cargo.toml

```
@@ -55,6 +55,10 @@ criterion = "0.5.0"
 # feature when testing, so `cargo test` works correctly.
 jpeg = { package = "jpeg-decoder", version = "0.3.0", default-features = false, features = ["platform_independent"] }
 
+[package.metadata.docs.rs]
+all-features = true
```

I think we have to avoid enabling the `avif` and `benchmarks` feature or we'll have build issues when docs.rs tries to compile the crate. The one feature we do want to enable however is rayon:

```suggestion
features = ["rayon"]
```

---

_Just having looked at the code for a tiny bit_

I don't think `benchmarks` should be an issue since everything that depends on `benchmark` is behind `#[cfg(test)]` anyway. A quick local rustdocs build confirmed that or am I missing something here?

Why is `aviv` a problem? Because it depends on the `dav1d` system lib? Tbh I don't really have experience with docs.rs and system libraries, but I would hope that sys libraries are not an issue for docs.rs, since there are many crates with system lib dependencies on docs.rs, right? That is just me assuming tho 😅, do you have any experience with that?

---

So looks like `dav1d` does not work with docs.rs ([see here](https://docs.rs/crate/dav1d/latest)), which is kind of a shame. I wish they would just handle the docs.rs environment in their build script, like suggested [here](https://docs.rs/about/builds#detecting-docsrs). Maybe I'll add a PR for the `dav1d-sys` crate in the next few days (don't know if I'll get to it tho).

Apart from that, there apparently is [a way to add system libraries to the docs.rs build environment](https://forge.rust-lang.org/docs-rs/add-dependencies.html) as well as [a way to test the build in the same environment as docs.rs](https://forge.rust-lang.org/docs-rs/add-dependencies.html#testing-the-image). I'll see if I can find some time to test it in the next few days, but again, I can't make any promises.